spec/unit/sensu_role_spec.rb

```diff
@@ -28,6 +28,21 @@
     }.to raise_error(Puppet::Error, 'Title or name must be provided')
   end
 
+  it 'should handle composite title' do
```

Comment:
what happens if a composite title and a namespace are specified? what happens if they are specified incorrectly, ie `foo in qa` is the title and the namespace is `default`.

what happens if the title is named funny? `in in qa` or `weird       spacing                   in qa` or they use hard tabs?

Reply:
For the first issue, `namespace` property takes priority but I'll add a test for that. So name `foo in qa` with `namespace` of `default` will keep namespace `default`.

If the pattern doesn't match it just puts the entire name in to `name` property and that becomes default for `resource_name` and `resource_name` doesn't allow spaces so those 2 examples would raise an error. We already test for this indirectly but I'll see if I can add some reasonable tests for this with compound aspect.

Reply:
Thanks! Since this introduces a regex on the title, we want to ensure we are testing the regex as well as the edge cases of people mixing composite naming with namespaces.

Mixing could happen inadvertently with Hiera pretty easily.

Reply:
Added some extra unit tests for composite names.